### PR TITLE
Update the geometry optimization function

### PR DESCRIPTION
**Pull request by ddhawan11**
**Context:**
Updated the geometry optimization function to perform a single task

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane/qchem/vibrational/pes_generator.py [hunk 1]
```diff
@@ -822,7 +822,8 @@ def vibrational_pes(
 
     comm = MPI.COMM_WORLD
     rank = comm.Get_rank()
-    molecule, scf_result = optimize_geometry(molecule, method)
+    molecule = optimize_geometry(molecule, method)
```
Comment (soranjh) [inline pennylane/qchem/vibrational/pes_generator.py:825]: The output should be an array not the molecule object, then you can reconstruct the molecule object with the new geometry here and pass it to _single_point

File: tests/qchem/vibrational/test_vibrational_class.py [hunk 2]
```diff
@@ -99,12 +99,12 @@ def test_single_point_energy(sym, geom, unit, method, basis, expected_energy):
...
             ["H", "F"],
             np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.0]]),
-            np.array([[0.0, 0.0, 0.07497201], [0.0, 0.0, 1.81475336]]),
+            np.array([[0.0, 0.0, 0.03967348], [0.0, 0.0, 0.96032612]]),
         ),
         (
             ["C", "O"],
             np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.0]]),
-            np.array([[0.0, 0.0, -0.12346543], [0.0, 0.0, 2.0131908]]),
+            np.array([[0.0, 0.0, -0.06533509], [0.0, 0.0, 1.06533469]]),
```
Comment (ddhawan11) [inline tests/qchem/vibrational/test_vibrational_class.py:107]: Changed these as earlier the results were always in Bohr.

File: pennylane/qchem/vibrational/vibrational_class.py [hunk 3]
```diff
@@ -152,9 +150,7 @@ def optimize_geometry(molecule, method="rhf"):
             Hartree-Fock,  ``'rhf'`` and ``'uhf'``, respectively. Default is ``'rhf'``.
 
     Returns:
-        tuple: A tuple containing the following:
-         - :func:`~pennylane.qchem.molecule.Molecule` object with optimized geometry
-         - pyscf.scf object
+        array[array[float]]: atomic coordinates at equilibrium
```
Comment (soranjh) [inline pennylane/qchem/vibrational/vibrational_class.py:153]:
```suggestion
        array[array[float]]: optimized atomic positions in Cartesian coordinates
```

**Review by soranjh — APPROVED** [review]
@thanks @ddhawan11.

**Review by austingmhuang — APPROVED** [review]
Make sure to address codecov before merging.

File: tests/qchem/vibrational/test_vibrational_class.py (hunk 2 above)
Comment (soranjh) [reply tests/qchem/vibrational/test_vibrational_class.py:107]: You need to test both Ang and Bohr, otherwise Codecov will complain.

**Pull request merged**